build.gradle

```diff
@@ -41,6 +52,110 @@ subprojects {
+            with jar
+        }
+        build.finalizedBy(fatJar)
+
+        task copyScripts(type: Copy){
+            mkdir "$projectDir/scripts/"
+            from "${projectDir}/../../k8s-deployment/common/"
+            into "$projectDir/scripts/"
+        }
+
+        task dockerfile(type: Dockerfile) {
+            dependsOn copyScripts
+            dependsOn fatJar
+            destFile = project.file('./Dockerfile')
+            from 'alpine:3.7'
+            label(['maintainer': 'huawei@huawei.com'])
+            runCommand  'apk --update add openjdk8-jre bind-tools'
+            // add this statement for respective Application
+            //copyFile './build/libs/${ProjectName}.jar', '/root/amino/jars/'
+            copyFile './scripts/*', '/root/amino/'
+        }
+
+        task buildDockerImage(type: DockerBuildImage) {
+            dependsOn dockerfile
+            url = 'unix:///var/run/docker.sock'
+            inputDir = file("$projectDir")
+        }
+
+        task pushImage(type: DockerPushImage) {
+            dependsOn buildDockerImage
+        }
+
+        task createOMSDeployment (type: Copy) {
+            from "${projectDir}/../../k8s-deployment/"
+            into "$projectDir"
+            include 'oms.template.yml'
+            rename { file -> 'oms.yml' }
+        }
+
+        task createKernelServerDeployment (type: Copy) {
+            from "${projectDir}/../../k8s-deployment/"
+            into "$projectDir"
+            include 'kernelserver.template.yml'
+            rename { file -> 'kernelserver.yml' }
+        }
+
+        deployer {
+            kubernetes {
+                id = project.property('omsId')
+                baseUrl = project.property('baseUrl')
+                namespace = project.property('namespace')
+                strategy = project.property('strategy')
+                auth {
+                    username = project.property('k8s.username')
+                    password = project.property('k8s.password')
+                }
+                ssl {
+                    trustAll = true
+                }
+                files = [file("oms.yml")]
+            }
+        }
+
+        deployer {
+            kubernetes {
+                id = project.property('kernelServerId')
+                baseUrl = project.property('baseUrl')
+                namespace = project.property('namespace')
+                strategy = project.property('strategy')
+                auth {
+                    username = project.property('k8s.username')
+                    password = project.property('k8s.password')
+                }
+                ssl {
+                    trustAll = true
+                }
+                files = [file("kernelserver.yml")]
+            }
+        }
+
+        task deployOms(type: Exec){
+            dependsOn pushImage
+            dependsOn createOMSDeployment
+            commandLine "${projectDir}/../../gradlew", 'deploy', "--environmentId=oms"
```

Comment:
I'm not that keen on gradle calling gradle.  Is there not a better way to do this?

Reply:
@quinton-hoole 
We have reused ```deployer``` task of ```de.qaware.cloud.deployer``` which was already present. ```deployer``` task is tagged with its ID. So based on IDs tasks are getting executed.  
Considering this, developer has to use "bash gradlew deploy --environmentId=OMS" himself to get OMS deployed.

We did find out an alternate approach in which we can use a library which has Kubernetes client (in java). We can use this client to write a task for OMS/KernelServer deployment. 

Please let me know if this approach looks ok for you.
May look like:

```
def myMethod = { param1, param2, param3, param4, param5 ->
            println("Hello World " + param1)
            Config config = new ConfigBuilder().withMasterUrl("http://127.0.0.1:32768").build()
            KubernetesClient client = new DefaultKubernetesClient(config)
            Deployment deployment1 = new DeploymentBuilder()
                    .withNewMetadata()
                        .withName("kernelserver-deployment")
                    .endMetadata()
                    .withNewSpec()
                        .withReplicas(1)
                        .withNewSelector()
                            .addToMatchLabels("app", "kernelserver")
                        .endSelector()
                        .withNewTemplate()
                            .withNewMetadata()
                                .addToLabels("app", "kernelserver")
                            .endMetadata()
                            .withNewSpec()
                                .addToContainers(new ContainerBuilder()
                                                    .withName("kernelserver")
                                                    .withImage("${param2}" + ":latest")
                                                    .withImagePullPolicy("Always")
                                                    .withPorts(Arrays.asList(new ContainerPortBuilder()
                                                                                .withContainerPort(32558)
                                                                                .withContainerPort(32557)
                                                                                .build()))
                                                    .addToCommand("/root/amino/start-kernelserver.sh")
                                                    .addToArgs("oms-svc-${param2}", "${param3}", "${param4}", "${param5}")
                                                    .addNewEnv()
                                                        .withName("MY_NODE_NAME")
                                                        .withNewValueFrom()
                                                            .withNewFieldRef()
                                                                .withNewFieldPath("spec.nodeName")
                                                            .endFieldRef()
                                                        .endValueFrom()
                                                    .endEnv()
                                                .build())
                            .endSpec()
                        .endTemplate()
                    .endSpec().build()
            client.apps().deployments().inNamespace("default")
                    .withName("kernelserver-deployment").create(deployment1)
            println("Square: " + Integer.parseInt(param1)*Integer.parseInt(param1))
        }
```

Reply:
@quinton-hoole 
We have implemented as per @AmitRoushan comment. 
Please have a look if it looks ok.
changes are in commit 0a251f3e93156a5b29f86919dc71a8b4fe93101f